cql3/functions/user_function.cc

```diff
@@ -58,7 +60,12 @@ bytes_opt user_function::execute(cql_serialization_format sf, const std::vector<
         },
         [&] (wasm::context& ctx) {
             try {
-                return wasm::run_script(ctx, arg_types(), parameters, return_type(), _called_on_null_input).get0();
+                auto func_cache = ctx.cache;
+                auto func_inst = func_cache->get(name(), ctx).get0();
+
+                return wasm::run_script(ctx, *func_inst.store, *func_inst.instance, *func_inst.func, arg_types(), parameters, return_type(), _called_on_null_input).then_wrapped([&, func_inst = std::move(func_inst)] (future<bytes_opt> f) mutable {
+                    return ctx.cache->recycle(name(), ctx, std::move(func_inst)).then([f = std::move(f)] () mutable { return std::move(f);});
+                }).get0();
```

Comment:
Why this combination of thread and continuation code? Keep one.

Reply:
I did this to fix an issue with compilation, but after writing it using only thread, it works too